pkg/httpapi/conversion.go

```diff
@@ -50,14 +51,27 @@ func applicationsToAppsResponse(appSet []*argoV1aplha1.Application, repoURL stri
 			continue
 		}
 
+		var lastDeployed metav1.Time
+		size := len(app.Status.History)
+		if size > 0 {
+			lastDeployed = app.Status.History[size-1].DeployedAt
+		}
+		var lastDeployedTime = lastDeployed.String()
+		if lastDeployed.IsZero() {
+			lastDeployedTime = ""
+		}
 		if appResp, ok := appsMap[appName]; !ok {
 			appsMap[appName] = appResponse{
 				Name:         appName,
 				RepoURL:      app.Spec.Source.RepoURL,
 				Environments: []string{app.Spec.Destination.Namespace},
+				SyncStatus:   []string{string(app.Status.Sync.Status)},
+				LastDeployed: []string{lastDeployedTime},
```

Comment:
I wonder if it would better to return empty slice when there is no deployment.  Easier for the UI to handle?

Reply:
I think it's easier to handle, especially if we have more than one environment for an app.  Here are examples:
{""}.  <-- as for this case here
{"", aValidTime1}
{aValidTime, "", aValidTime2}
....
It's more consistent this way.
The front end handles this easily.

Reply:
It seems to me that not adding an entry to slice when there is no deployments is easier for the client but I guess the client already handles it like you say.  Then, it is fine.  

Does an old client (4.7) handle these extra fields gracefully?   We've spoken about adding a version number field in the resquest header.   Now that client's frozen, if we added a verson header it would probably get ignored anyway.  We can still add it to the api for future clients to take advantage of it.  WDYT?

Reply:
The old client 4.7 is not affect because it still calls the old /pipelines api.  Eventually, this api will be deprecated and removed.

This new client 4.8, against an older backend without the new List API, will handle it gracefully.  The beauty of this is that the sync status and last deployed 'fields' are made optional.  Shubham and I briefly talked about the version number field in the request header, but for this, it was not needed. It will just work.  The app list table will resort to showing the list of environments in place of the sync status, and there will be no `last deployment` times ( '-' is shown instead).   

Basically, the front end needs to handle an error condition for which something it expects to be there is not there, whether it is 'version' or a new API. There is no pressing need to add the version to the header and I think we could revisit this when we cross that bridge again.